aggregators/aggregator.go

```diff
@@ -603,23 +603,134 @@ func (a *Aggregator) processHarvest(
 	hs := harvestStats{
 		eventsTotal:            cm.EventsTotal,
 		youngestEventTimestamp: modelpb.ToTime(cm.YoungestEventTimestamp),
-		servicesOverflowed:     hllSketchEstimate(cm.OverflowServicesEstimator),
 	}
-	addOverflow := func(o *aggregationpb.Overflow) {
+
+	a.getAndLogOverflowStats(&hs, cm, aggIvl)
+
+	if err := a.cfg.Processor(ctx, cmk, cm, aggIvl); err != nil {
+		return harvestStats{}, fmt.Errorf("failed to process combined metrics ID %s: %w", cmk.ID, err)
+	}
+	return hs, nil
+}
+
+func (a *Aggregator) getAndLogOverflowStats(hs *harvestStats, cm *aggregationpb.CombinedMetrics, aggIvl time.Duration) {
+	hs.servicesOverflowed = hllSketchEstimate(cm.OverflowServicesEstimator)
+
+	logFunc := func(msg string, fields ...zap.Field) {
+		if a.cfg.OverflowLogging.Func != nil &&
+			(a.cfg.OverflowLogging.AggregationInterval == 0 || a.cfg.OverflowLogging.AggregationInterval == aggIvl) {
+			a.cfg.OverflowLogging.Func(msg, fields...)
+		}
+	}
```

Comment:
If we go with the other suggestion to always either log at warning level, or don't log at all, then I think this would become:

```go
var logger *zap.Logger
if a.cfg.overflowLoggingEnabled {
	logger = a.cfg.Logger.With(zap.String(aggregationIvlKey, formatDuration(aggIvl)))
} else {
	logger = zap.NewNop()
}
```